### PR TITLE
Changed name of autism pin and gold autism pin

### PR DESCRIPTION
**Pull request by foob1**
## About the PR
Changed the autism pin to be called the neurodiversity pin, and changed the gold autism pin to be called the autism pin

## Why / Balance
While the current sprite for the autism pin is for autism, its more commonly used for neurodiversity. Changing it to be for neurodiversity allows for more representation of all neurodivergent people instead of just autism. The gold autism pin however is used specifically just for autism, which is why it is renamed to be the autism pin.

## Media
image
image
image

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**
:cl:
- tweak: Changed the names of the autism pin and gold autism pin

### CODE REVIEW

**Review by lzk228 — COMMENTED** [review]
add changed ids to migration.yml

File: Resources/migration.yml [hunk 1]
```diff
@@ -406,3 +406,7 @@ FoodMeatFiestaKebab: FoodKebabSkewer
 
 # 2024-08-19
 ClothingNeckShockCollar: ClothingBackpackElectropack
+
+# 2024-10-13
+ClothingNeckGoldAutismPin: ClothingNeckAutismPin
+ClothingNeckAutismPin: ClothingNeckNeurodiversityPin
```
Comment (lzk228) [inline Resources/migration.yml:412]:
```suggestion
ClothingNeckAutismPin: ClothingNeckNeurodiversityPin

```

File: Resources/Prototypes/Entities/Clothing/Neck/pins.yml [hunk 2]
```diff
@@ -124,22 +124,22 @@
...
+  id: ClothingNeckAutismPin
+  name: autism pin
+  description: Be autism do crime.
   components:
   - type: Sprite
-    state: goldautism
+    state: autism
   - type: Clothing
-    equippedPrefix: goldautism
+    equippedPrefix: autism
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Clothing/Neck/pins.yml:145]: weh
```suggestion
    equippedPrefix: autism

```

File: Resources/Textures/Clothing/Neck/Misc/neurodiversitypin.rsi/meta.json
Comment (lzk228) [inline Resources/Textures/Clothing/Neck/Misc/neurodiversitypin.rsi/meta.json:1]: i don't understand, why did you create a new rsi file? you just need to change names and states in yml

Comment (foob1) [reply Resources/Textures/Clothing/Neck/Misc/neurodiversitypin.rsi/meta.json:1]: I'll be honest I don't really know, I did this months ago and I'm just picking it up again. Whatever it is, it works when I test it... unsure about the test fails though.

Comment (lzk228) [reply Resources/Textures/Clothing/Neck/Misc/neurodiversitypin.rsi/meta.json:1]: try to do it from the scratch

**Comment by foob1** [comment]
Closing this PR, maybe I'll try again and do it from scratch

**Pull request closed without merging**